app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.java

```diff
@@ -355,6 +356,32 @@ void mobileDataDialog() {
     flag = false;
   }
 
+  @Override public void hotspotTurnedOn(WifiConfiguration wifiConfiguration) {
+
+    //Show an alert dialog for hotspot details
+    AlertDialog.Builder builder = new AlertDialog.Builder(this, dialogStyle());
+    WebServerHelper webServerHelper = new WebServerHelper(this);
+    builder.setPositiveButton(android.R.string.ok, (dialog, id) -> {
+      final Handler handler = new Handler();
+      handler.postDelayed(new Runnable() {
+        @Override
+        public void run() {
+          webServerHelper.startServerHelper();
+        }
+      }, 2000);
+    });
+
+    builder.setTitle(this.getString(R.string.hotspot_turned_on));
+    builder.setMessage(
+        this.getString(R.string.hotspot_details_message) + "\n" + this.getString(
+            R.string.hotspot_ssid_label) + " " + wifiConfiguration.SSID + "\n" + this.getString(
+            R.string.hotspot_pass_label) + " " + wifiConfiguration.preSharedKey);
+
+    builder.setCancelable(false);
+    AlertDialog dialog = builder.create();
+    dialog.show();
```

Comment:
@macgills I've shifted the dialog to `ZImHostActivity`. Now, when I'm going away from the `ZimHostActivity` and then coming back and trying to start dialog. I get this error:

```
2019-08-07 17:16:24.818 19041-19041/org.kiwix.kiwixmobile E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.kiwix.kiwixmobile, PID: 19041
    android.view.WindowManager$BadTokenException: Unable to add window -- token android.os.BinderProxy@8e8e9b5 is not valid; is your activity running?
        at android.view.ViewRootImpl.setView(ViewRootImpl.java:958)
        at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:390)
        at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:93)
        at android.app.Dialog.show(Dialog.java:379)
        at org.kiwix.kiwixmobile.webserver.ZimHostActivity.hotspotTurnedOn(ZimHostActivity.java:382)
        at org.kiwix.kiwixmobile.wifi_hotspot.WifiHotspotManager$1.onStarted(WifiHotspotManager.java:55)
        at android.net.wifi.WifiManager$LocalOnlyHotspotCallbackProxy$1.handleMessage(WifiManager.java:2832)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:226)
        at android.app.ActivityThread.main(ActivityThread.java:7179)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:576)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:942)
```

Reply:
@macgills I'm not sure why it is unable to show the dialog from the activity itself now.

Reply:
`is your activity running?` I imagine it may be getting called before onCreate happens